### PR TITLE
Implementation Considerations for Deployments and Products

### PR DESCRIPTION
**Pull request by tlodderstedt**

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -627,11 +627,33 @@ Alternatively, there could be an authorization data type for OpenID Connect. (#o
...
+
+* Support advertisement of supported authorization details types in OAuth server metadata
+* Accept `authorization_details` parameter in authorization requests including basic syntax check for compliance with this specification 
+* Support storage of consented authorization_details as part of a grant
+* Implement default behavior for carrying authorization details to resource servers access token content and introspection responses. If the product supports resource indicators, this should also include assignment of authorization details to access tokens based on `resource` token request parameter
+
+Processing and presentation of authorization details will vary significantly among different authorization data types. Products should therefore support customization of the respective behavior. In particular products should 
+  
+* allow deployments to determine presentation of the authorization_details
+* allow deployments to modify requested authorization_details in the user consent process, e.g. adding fields
```
Comment (vdzhuvinov) [inline main.md:654]: Could you give an example of one such modification?
Is that about applying default values to the authorization_details before it gets presented to the user?

Comment (tlodderstedt) [reply main.md:654]: That’s one option and an example of the AS applying a policy.

Another example is values determined by the user: The client might ask for access to account balance and the user picks the accounts the client shall have access to. It’s described in more detail in the spec: https://tools.ietf.org/html/draft-ietf-oauth-rar-03#section-3.4.1.

Comment (vdzhuvinov) [reply main.md:654]: Thanks Torsten. So that is related to the "Enrich authorization details..." bullet.

Comment (tlodderstedt) [reply main.md:654]: yes

Comment (tlodderstedt) [reply main.md:654]: Based on recent discussions around the grant management API, I could also imagine the AS to add expiration times and/or identifiers to authorization details objects.

**Comment by tlodderstedt** [comment]
@jricher @b---c I would like to merge this PR. Ok with you?

**Review by jricher — APPROVED** [review]
Seems like it might be overreach in terms of guidance for implementors, but that can be walked back if needed.

**Comment by b---c** [comment]
I still maintain that this presumes a specific product and deployment architecture that, while It may well be a common one, is not exclusive. That could maybe be qualified or explained a little better. But I'm okay with it merging regardless.

**Review by b---c — APPROVED** [review]

**Comment by vdzhuvinov** [comment]
We'll be doing some RAR use case modelling and after that I'll submit concrete feedback / comments if there are any.

**Comment by tlodderstedt** [comment]
@b---c Can you please point out where you perceive such "a specific product and deployment architecture"? I tried to avoid such assumptions and instead described what a deployment shall be able to influence/customize.

**Comment by tlodderstedt** [comment]
@vdzhuvinov thanks a lot in advance! The question for me is whether we shall postpone merge of this PR as long as your are working on the use cases or whether we will include your feedback in a subsequent revision of the draft. How long will the process take on your side?

**Comment by vdzhuvinov** [comment]
Feel free it merge the PR, it will take some time do this work here.

**Comment by bc-pi** [comment]
I don't really know how to articulate it any better and am okay with proceeding with the merge anyway.

**Pull request merged**